etna/datasets/internal_datasets.py

```diff
@@ -0,0 +1,135 @@
+import os
+import tempfile
+import urllib.request
+import warnings
+import zipfile
+from pathlib import Path
+from typing import Dict
+
+import pandas as pd
+
+from etna.datasets import TSDataset
+
+_DOWNLOAD_PATH = Path.home() / ".etna"
+
+
+def _check_dataset_local(dataset_path: Path) -> bool:
+    """
+    Check dataset is local.
+
+    Parameters
+    ----------
+    dataset_path:
+        path to dataset
+    """
+    return os.path.isfile(dataset_path)
+
+
+def _download_dataset_zip(url: str, file_name: str, **kwargs) -> pd.DataFrame:
+    """
+    Download zipped csv file.
+
+    Parameters
+    ----------
+    url:
+        url of the dataset
+    file_name:
+        csv file name in zip archive
+
+    Returns
+    -------
+    result:
+        dataframe with data
+
+    Raises
+    ------
+    Exception:
+        any error during downloading, saving and reading dataset from url
+    """
+    try:
+        with tempfile.TemporaryDirectory() as td:
+            temp_path = os.path.join(td, "temp.zip")
+            urllib.request.urlretrieve(url, temp_path)
+            with zipfile.ZipFile(temp_path) as f:
+                f.extractall(td)
+                df = pd.read_csv(os.path.join(td, file_name), **kwargs)
+    except Exception as err:
+        raise Exception(f"Error during downloading and reading dataset. Reason: {repr(err)}")
+    return df
+
+
+def load_dataset(name: str, download_path: Path = _DOWNLOAD_PATH, rebuild_dataset: bool = False) -> TSDataset:
+    """
+    Load internal dataset.
+
+    Parameters
+    ----------
+    name:
+        Name of the dataset.
+    download_path:
+        The path for saving dataset locally.
+    rebuild_dataset:
+        Whether to rebuild dataset from the initial source.
```

Comment:
I find it difficult to understand that this option is related to caching. We should reformulate it or give more explanations.